NumLib/Fem/ShapeFunction/ShapeQuad4-impl.h

```diff
@@ -16,23 +16,24 @@ namespace NumLib
 template <class T_X, class T_N>
 void ShapeQuad4::computeShapeFunction(const T_X &r, T_N &N)
 {
-    N[0] = (1.0 + r[0]) * (1.0 + r[1]) / 4;
-    N[1] = (1.0 - r[0]) * (1.0 + r[1]) / 4;
-    N[2] = (1.0 - r[0]) * (1.0 - r[1]) / 4;
-    N[3] = (1.0 + r[0]) * (1.0 - r[1]) / 4;
+    N[0] = (1.0 - r[0]) * (1.0 - r[1]) / 4;
+    N[1] = (1.0 + r[0]) * (1.0 - r[1]) / 4;
+    N[2] = (1.0 + r[0]) * (1.0 + r[1]) / 4;
+    N[3] = (1.0 - r[0]) * (1.0 + r[1]) / 4;
 }
```

Comment:
It does matter, because now the solution is correct. The ordering of N_u and N_p shape functions must correspond, otherwise products like dN_u / dx * N_p are wrong.